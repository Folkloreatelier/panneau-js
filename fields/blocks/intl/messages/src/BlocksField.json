[
  {
    "id": "fields.blocks.add_button_type_label",
    "description": "The block type label on the add button",
    "defaultMessage": "a block"
  }
]