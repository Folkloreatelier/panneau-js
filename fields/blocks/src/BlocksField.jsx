import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import get from 'lodash/get';
import { PropTypes as PanneauPropTypes } from '@panneau/core';
import ItemsField from '@panneau/field-items';
import { defineMessages } from 'react-intl';

const messages = defineMessages({
    typeLabel: {
        id: 'fields.blocks.add_button_type_label',
        description: 'The block type label on the add button',
        defaultMessage: 'a block',
    },
});

const propTypes = {
    addButtonTypeLabel: PanneauPropTypes.message,
};

const defaultProps = {
    addButtonTypeLabel: messages.typeLabel,
};

const contextTypes = {
    definition: PanneauPropTypes.definition,
};

class BlocksField extends Component {
    getTypes() {
        const { definition } = this.context || {};
        return get(definition || null, 'blocks', null);
    }
    render() {
        return <ItemsField {...this.props} types={this.getTypes()} />;
    }
}

BlocksField.propTypes = propTypes;
BlocksField.defaultProps = defaultProps;
BlocksField.contextTypes = contextTypes;

export default BlocksField;
