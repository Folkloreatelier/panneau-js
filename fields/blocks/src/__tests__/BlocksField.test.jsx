/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import BlocksField from '../BlocksField';

test('match snapshot', () => {
    const component = renderer.create((
        <IntlProvider locale="en">
            <BlocksField
                label="Label"
                value={[]}
            />
        </IntlProvider>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
