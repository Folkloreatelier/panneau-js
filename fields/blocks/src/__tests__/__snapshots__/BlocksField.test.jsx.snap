// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="form-group undefined"
>
  <div
    className="form-group-label"
  >
    <label
      className="control-label"
      htmlFor="fields"
    >
      Label
    </label>
  </div>
  <div
    className="form-group-inner"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className=""
    >
      <div
        className="btn-group"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          className="btn btn-default"
          data-toggle="dropdown"
          onClick={[Function]}
          type="button"
        >
          <span>
            Add a block
          </span>
        </button>
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className="list"
        lockAxis="y"
        onSortEnd={[Function]}
        useDragHandle={true}
      >
        <div
          className={undefined}
        />
      </div>
    </div>
  </div>
</div>
`;
